core/src/main/java/io/cucumber/core/feature/GluePath.java

```diff
@@ -29,6 +34,11 @@
  */
 public class GluePath {
 
+    private static final Logger log = LoggerFactory.getLogger(GluePath.class);
+
+    private static final Pattern FILESYSTEM_PATH_TO_PACKAGES = Pattern
+            .compile("src/(?:main|test)/(?:java|kotlin|scala|groovy)(|/|/.+)");
```

Comment:
By having the capture group match either `''`, `'/'` or '/anything else'` we can avoid checking a few edge cases in the warning. We now know the string is either empty or starts with a `/`.